_layouts/default.html

```diff
@@ -18,7 +18,7 @@
   <body>
     <header>
       <div class="inner">
-        <h1>{{ site.title | default: site.github.repository_name }}</h1>
+        <h1><a href="{{ site.baseurl }}">{{ site.title | default: site.github.repository_name }}</a></h1>
```

Comment:
> `site.baseurl`

I would use `"/" | absolute_url` to let Jekyll handle the logic (users do all sorts of crazy things with `baseurl`).